app/scripts/controllers/network/createJsonRpcClient.js

```diff
@@ -7,19 +8,34 @@ import createBlockTrackerInspectorMiddleware from 'eth-json-rpc-middleware/block
 import providerFromMiddleware from 'eth-json-rpc-middleware/providerFromMiddleware'
 import BlockTracker from 'eth-block-tracker'
 
+const inTest = process.env.IN_TEST === 'true'
+const blockTrackerOpts = inTest
+  ? { pollingInterval: 1000 }
```

Comment:
Before this PR, these opts were not just included when `inTest` https://github.com/MetaMask/metamask-extension/pull/9551/files#diff-5d7a765d6c410d375ee6359517ee8333L14

Why is that change happening now?

Reply:
Previously we were using the old localhost provider in test, which did have this option set. See below, the deleted file `createLocalHostClient.js`.

The only change here is setting the polling time to 1s instead of the default of 20s when in test.

Reply:
The link doesn't take me to a specific line, but I assume you're referring to the content of `createLocalhostClient`: https://github.com/MetaMask/metamask-extension/blob/6409caa0814d228e42b005992b96b55b92948431/app/scripts/controllers/network/createLocalhostClient.js

To answer your question, `createLocalhostClient` was basically `createJsonRpcClient`, with some optimizations/configuration options for `localhost` networks. We don't _need_ to make those optimizations, so now we ignore them in production.

However, our e2e tests rely on those optimizations. The [default polling interval is 20 seconds](https://github.com/MetaMask/eth-block-tracker/blob/master/src/polling.js#L14), which is longer than many of our e2e test timeouts.

I haven't confirmed whether the estimate gas delay or polling interval alone are sufficient to get e2e tests reliably passing, but e2e tests do fail in the absence of both. Given how flaky our e2e tests are, the safest bet is probably to include both options.

Reply:
Now, this will degrade development experience somewhat on Ganache networks (since the polling interval is the same as on a "real" network), but we can address that after 8.1.

Reply:
Keeping polling rate more dynamic is already a good thing for us to aim for, as different EVM blockchains have different block frequencies.

[Opened an issue](https://github.com/MetaMask/metamask-extension/issues/9556)